ta/pkcs11/src/pkcs11_attributes.c

```diff
@@ -1338,6 +1364,9 @@ enum pkcs11_rc check_created_attrs(struct obj_attrs *key1,
 	}
 	if (private) {
 		switch (get_key_type(private)) {
+		case PKCS11_CKK_RSA:
+			/* public key template carries bit size */
```

Comment:
In addition to RSA also EC and generic secret key gen output bits from `pkcs11_mechanism_supported_key_sizes()` which can be confusing. That method is used to supply information to `C_GetMechanismInfo()` and is specified as:
`size of the key for the mechanism (whether this is measured in bits or in bytes is mechanism-dependent)`. So I would not change implementation of `pkcs11_mechanism_supported_key_sizes()`.

With EC they are not easily divisible with 8. In theory RSA allows key sizes which are not aligned to 8 bits (but not used in practice as far as I know).

`mechanism_supported_key_sizes_bytes()` does have fixup for `PKCS11_CKM_GENERIC_SECRET_KEY_GEN` so we could have `switch` for all types that needs to have bits converted to bytes and ceiled.

This would then directly provide output to `get_key_min_max_sizes()`.

Also `mechanism_supported_key_sizes_bytes()` needs to be aware of EC for that to work so need to add support for that in `get_key_min_max_sizes()`.

Does that sound good?

Reply:
sounds good to me